Add PV.Name to volume tags.

jsafrane opened the pull request:
We add claim.Name and claim.Namespace as tags to AWS EBS / GCE PD / OpenStack
Cinder volumes created by Kubernetes. To easily match Kubernetes volumes and
cloud volumes, let's add also PV.Name.

This allows easier matching between Kubernetes and real cloud volumes.

jsafrane commented:
cc: @kubernetes/rh-storage

markturansky commented on pkg/controller/persistentvolume/persistentvolume_provisioner_controller.go:
```diff
@@ -330,15 +330,19 @@ func (controller *PersistentVolumeProvisionerController) Stop() {
+	tags := make(map[string]string)
+	tags[cloudVolumeCreatedForClaimNamespaceTag] = claim.Namespace
+	tags[cloudVolumeCreatedForClaimNameTag] = claim.Name
+	if pv != nil {
```
When would PV not be nil?  We first make the PV from the template and persist it with ClaimRef to the PVC.

It seems like nil PV here should be an error.

jsafrane replied:
`newProvisioner` is called twice and at the first call the PV is not yet availabe, it's created by the returned provisioner.

https://github.com/kubernetes/kubernetes/pull/18957/files#diff-fc923807bec23504369a29c0f94b6055R175

derekwaynecarr replied:
can you add a comment to that effect?  I am less familiar with this area when doing the review and had a similar thought.

jsafrane replied:
I added a short note.

jsafrane committed:
Add PV.Name to volume tags.

We add claim.Name and claim.Namespace as tags to AWS EBS / GCE PD / OpenStack
Cinder volumes created by Kubernetes. To easily match Kubernetes volumes and
cloud volumes, let's add also PV.Name.

derekwaynecarr commented:
Is there no test case that can verify the tags on a pvc?  Can we add a mock unit test?

jsafrane commented:
> Is there no test case that can verify the tags on a pvc? Can we add a mock unit test?

I'll try to squeeze it into some provisioning test.

jsafrane committed:
Add an integration test for volume tags.

jsafrane commented:
Test added.

thockin commented on pkg/controller/persistentvolume/types.go:
```diff
@@ -32,10 +32,13 @@ const (
 	// Name of a tag attached to a real volume in cloud (e.g. AWS EBS or GCE PD)
 	// with namespace of a persistent volume claim used to create this volume.
-	cloudVolumeCreatedForNamespaceTag = "kubernetes.io/created-for/pvc/namespace"
+	cloudVolumeCreatedForClaimNamespaceTag = "kubernetes.io/created-for/pvc/namespace"
```
This should fail validation - only one / is allowed

thockin replied:
Sorry, I interpreted this as a k8s label, not a cloud tag

thockin commented:
add to whitelist

markturansky commented:
@k8s-bot test this

markturansky commented:
```

!!! [0113 17:42:19] Timed out waiting for apiserver:  to answer at http://127.0.0.1:8080/healthz; tried 120 waiting 1 between each
!!! Error in ./hack/test-update-storage-objects.sh:48
  'return 1' exited with status 1
Call stack:
```

jsafrane commented:
@k8s-bot test this

jsafrane commented:
All green. @derekwaynecarr , thanks for review.
